### PR TITLE
feat(demo): Update demo to work on OpenShift

### LINKED ISSUE #2744: Automated demo does not work on OpenShift
**Bug description**:
Automated demo doesn't work on OpenShift as OpenShift already uses port 80.

**Affected area (please mark with X where applicable)**:
- Install                    [ ]
- SMI Traffic Access Policy  [ ]
- SMI Traffic Specs Policy   [ ]
- SMI Traffic Split Policy   [ ]
- Permissive Traffic Policy  [ ]
- Ingress                    [ ]
- Egress                     [ ]
- Envoy Control Plane        [ ]
- CLI Tool                   [ ]
- Metrics                    [ ]
- Certificate Management     [ ]
- Sidecar Injection          [ ]
- Logging                    [ ]
- Debugging                  [ ]
- Tests                      [ ]
- CI System                  [ ]

**Expected behavior**:
Automated demo should work on OpenShift.

**Steps to reproduce the bug (as precisely as possible)**:
Run automated demo on an OpenShift cluster that uses port 80 for the OpenShift router (default config)

**How was OSM installed?**:
Automated demo
**Anything else we need to know?**:

**Environment**:
- OSM version (use `osm version`): 0.7.0
- Kubernetes version (use `kubectl version`): 1.18
- Size of cluster (number of worker nodes in the cluster): 
- Others:

### PR DESCRIPTION
**Pull request by ksubrmnn**
Signed-off-by: Kalya Subramanian <kasubra@microsoft.com>

**Description**: 
This PR updates the ports in the automatic and manual demos to use a "random" port that will not be reserved by OpenShift.

In the future, we can make the port configurable.

Fixes #2744

**Affected area**:

- New Functionality      [ ]
- Documentation          [ ]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Performance            [ ]
- Other                  [X]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
No

### CODE REVIEW

File: demo/cmd/common/books.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ const (
 	RestockWarehouseURL = "restock-books"
 
 	// bookstorePort is the bookstore service's port
-	bookstorePort = 80
+	bookstorePort = 8080
```
Comment (shashankram) [inline demo/cmd/common/books.go:22]: I know this is a work in progress, but we shouldn't be changing to default demo ports to 8080 just for this. If for some reason this port is unusable for Openshift, we should simply override the port similar to other configurable options.

In the past we did change the demo ports to 8080 and back to 80, more details can be found here https://github.com/openservicemesh/osm/issues/1099#issuecomment-660197168

File: demo/deploy-bookbuyer.sh [hunk 2]
```diff
@@ -21,6 +22,11 @@ metadata:
   namespace: $BOOKBUYER_NAMESPACE
 EOF
 
+if [ "$DEPLOY_ON_OPENSHIFT" = true ] ; then
+    oc adm policy add-scc-to-user privileged -z bookbuyer -n "$BOOKBUYER_NAMESPACE"
```
Comment (shashankram) [inline demo/deploy-bookbuyer.sh:26]: Need to document in the demo readme the dependency for `oc`.

**Review by shashankram — APPROVED** [review]

File: demo/deploy-bookbuyer.sh (hunk 2 above)
Comment (ksubrmnn) [reply demo/deploy-bookbuyer.sh:26]: Added to demo/README.md

**Review by shalier — APPROVED** [review]

**Review by snehachhabria — APPROVED** [review]

**Comment by ksubrmnn** [comment]
> I'll miss my goode olde port 80

To be addressed in https://github.com/openservicemesh/osm/issues/2755 :)

File: demo/cmd/common/books.go (hunk 1 above)
Comment (ksubrmnn) [reply demo/cmd/common/books.go]: Updated in favor of using a different port

**Pull request merged**